CMakeLists.txt

```diff
@@ -799,35 +799,37 @@ endif ()
 
 # Determine if a threading package is available on this system
 option (HDF5_ENABLE_THREADS "Enable thread support" ON)
```

Comment:
This option should be removed and the line replaces with
`set (H5_HAVE_THREADS 0)`

Then in src/CMakeLists.txt should remove the generator expressions involving the option.